src/yanfeishao/cs555/utils/OutputUtils.java

```diff
@@ -80,8 +81,37 @@ public void outputError(SimpleDBUtils simpleDBUtils, String prefix) {
     public Set<String> parseError(SimpleDBUtils simpleDBUtils, String prefix) {
         Set<String> result = new HashSet<>();
         switch (prefix) {
-            case ErrorCode.US01:
-                break;
+            case ErrorCode.US01: {
+                // Dates before the current date
+                Date todayDate = Calendar.getInstance().getTime();
+                simpleDBUtils.getFamilyDBList().forEach(familyEntity -> {
+                    Date marriageDate = familyEntity.getMarriedDate();
+                    Date divorceDate = familyEntity.getDivorceDate();
+                    Date husbandBirthDate = familyEntity.getFather().getBirthDate();
+                    Date wifeBirthDate = familyEntity.getMother().getBirthDate();
+                    Date husbandDeathDate = familyEntity.getFather().getDeathDate();
+                    Date wifeDeathDate = familyEntity.getMother().getDeathDate();
+                    if (husbandBirthDate != null && husbandBirthDate.after(todayDate) || husbandDeathDate != null && husbandDeathDate.after(todayDate)) {
```

Comment:
I think the logical should be if ((...&&...) || (...&&...)) over if (...&&....||....&&...||)

If you complete this modification, I will merge it to branch, although I do not think current design is good enough, when I have time I will look at it to see is there any way to improve the code style